src/python/WMComponent/RucioInjector/RucioInjectorPoller.py

```diff
@@ -341,69 +345,74 @@ def deleteBlocks(self):
+        for blocksSlice in grouper(blockDict, self.delBlockSlicesize):
+            logging.info("Handeling %d candidate blocks", len(blocksSlice))
+            containerDict = {}
+            # Populate containerDict, assigning each block to its correspondant container
+            for blockName in blocksSlice:
+                container = blockDict[blockName]['dataset']
+                # If the container is not in the dictionary, create a new entry for it
+                if container not in containerDict:
+                    # Set of sites to which the container needs to be transferred
+                    sites = set(x.replace("_MSS", "_Tape") for x in blockDict[blockName]['sites'])
+                    containerDict[container] = {'blocks': [], 'rse': sites}
+                containerDict[container]['blocks'].append(blockName)
+
+            blocksToDelete = []
+            for contName in containerDict:
+                cont = containerDict[contName]
+
+                # Checks if the container is not requested in any sites.
+                # This should never be triggered, but better safe than sorry
+                if not cont['rse']:
+                    logging.warning("No rules for container: %s. Its blocks won't be deleted.", contName)
+                    continue
 
-            try:
-                # Get RSE in which each block is available
-                availableRSEs = self.rucio.getReplicaInfoForBlocks(block=cont['blocks'])
-            except Exception as exc:
-                msg = "Failed to get replica info for blocks in container: %s.\n" % contName
-                msg += "Will retry again in the next cycle. Error: %s" % str(exc)
-                logging.error(msg)
-                continue
+                try:
+                    # Get RSE in which each block is available
+                    availableRSEs = self.rucio.getReplicaInfoForBlocks(block=cont['blocks'], deep=self.useDsetReplicaDeep)
+                except Exception as exc:
+                    msg = "Failed to get replica info for blocks in container: %s.\n" % contName
+                    msg += "Will retry again in the next cycle. Error: %s" % str(exc)
+                    logging.error(msg)
+                    continue
 
-            for blockRSEs in availableRSEs:
-                # If block is available at every RSE its container needs to be transferred, the block can be deleted
-                blockSites = set(blockRSEs['replica'])
-                if cont['rse'].issubset(blockSites):
-                    blocksToDelete.append(blockRSEs['name'])
+                for blockRSEs in availableRSEs:
+                    # If block is available at every RSE its container needs to be transferred, the block can be deleted
+                    blockSites = set(blockRSEs['replica'])
+                    logging.debug("BlockName: %s", blockRSEs['name'])
+                    logging.debug("Needed: %s / Available: %s", (str(cont['rse']), str(blockSites) ))
```

Comment:
This is an error. You either use a string substitution operator and a tuple while creating the log message, or put all sub strings needed in the logger's variable argument list and let it do the string expansion for you. So this one should be one  of those:
```
logging.debug("Needed: %s / Available: %s" % (str(cont['rse']), str(blockSites) ))
```
or
```
logging.debug("Needed: %s / Available: %s", str(cont['rse']), str(blockSites))
```

Reply:
yes,  pylint was complaining abour the first, so I  tried to change it to the second and missed the parenthesis. I'll fix it.